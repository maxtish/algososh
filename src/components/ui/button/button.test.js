import { Button } from "./button";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Проверяем Button при помощи снэпшотов корректную отрисовку", () => {
  it("кнопки с текстом", () => {
    const buttonWithText = renderer.create(<Button text="text" />);
    expect(buttonWithText).toMatchSnapshot();
  });

  it("кнопки без текста", () => {
    const buttonWithoutText = renderer.create(<Button />);
    expect(buttonWithoutText).toMatchSnapshot();
  });

  it("заблокированной кнопки", () => {
    const buttonDisabled = renderer.create(<Button disabled={true} />);
    expect(buttonDisabled).toMatchSnapshot();
  });

  it("кнопки с индикацией загрузки", () => {
    const buttonWithLoadingIndication = renderer.create(
      <Button isLoader={true} />
    );
    expect(buttonWithLoadingIndication).toMatchSnapshot();
  });

  describe("Проверяем корректность вызова колбека при клике на кнопку", () => {
    it("вызов колбека при клике на кнопку корректен", () => {
      const callBack = jest.fn();
      render(<Button onClick={callBack} />);
      const button = screen.getByTestId("button-test");
      fireEvent.click(button);
      expect(callBack).toHaveBeenCalled();
    });
  });
});
