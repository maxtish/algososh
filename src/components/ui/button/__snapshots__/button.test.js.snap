// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяем Button при помощи снэпшотов корректную отрисовку заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button-test"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Проверяем Button при помощи снэпшотов корректную отрисовку кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button-test"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Проверяем Button при помощи снэпшотов корректную отрисовку кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="button-test"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Проверяем Button при помощи снэпшотов корректную отрисовку кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button-test"
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
